cfme-performance/tests/workloads/test_provisioning.py

```diff
@@ -0,0 +1,97 @@
+"""Runs Provisioning Workload."""
+from utils.appliance import clean_appliance
+from utils.appliance import get_server_roles_workload_provisioning
+from utils.appliance import set_server_roles_workload_provisioning
+from utils.appliance import wait_for_miq_server_ready
+from utils.grafana import get_scenario_dashboard_url
+from utils.log import logger
+from utils.providers import add_providers
+from utils.providers import retire_provisioned_vms
+from utils.providers import provision_vm
+from utils.smem_memory_monitor import SmemMemoryMonitor
+from utils.ssh import SSHClient
+from utils.workloads import get_provisioning_scenarios
+from itertools import cycle
+import time
+import pytest
+
+
+@pytest.mark.parametrize('scenario', get_provisioning_scenarios())
+def test_provisioning(request, scenario):
+    """Runs through provisioning scenarios using the REST API to
+    continously order a catalog item(s) and provision a VM for a specified period of time.
+    Memory Monitor creates graphs and summary at the end of each scenario."""
+
+    from_ts = int(time.time() * 1000)
+    ssh_client = SSHClient()
+    logger.debug('Scenario: {}'.format(scenario['name']))
+
+    clean_appliance(ssh_client)
+
+    monitor_thread = SmemMemoryMonitor(SSHClient(), 'workload-provisioning', scenario['name'],
+        'Provisioning', get_server_roles_workload_provisioning(separator=', '),
+        ', '.join(scenario['providers']))
+
+    def cleanup_workload(scenario, from_ts):
+        starttime = time.time()
+        to_ts = int(starttime * 1000)
+        g_url = get_scenario_dashboard_url(scenario, from_ts, to_ts)
+        logger.debug('Started cleaning up monitoring thread.')
+        monitor_thread.grafana_url = g_url
+        monitor_thread.signal = False
+        monitor_thread.join()
+        retire_provisioned_vms(scenario['vm_names'])
+        timediff = time.time() - starttime
+        logger.info('Finished cleaning up monitoring thread in {}'.format(timediff))
+    request.addfinalizer(lambda: cleanup_workload(scenario, from_ts))
+
+    monitor_thread.start()
+
+    wait_for_miq_server_ready(poll_interval=2)
+    set_server_roles_workload_provisioning(ssh_client)
+    add_providers(scenario['providers'])
+    logger.info('Sleeping for Refresh: {}s'.format(scenario['refresh_sleep_time']))
+    time.sleep(scenario['refresh_sleep_time'])
+
+    cleanup_size = scenario['cleanup_size']
+    vm_names = []
+    for i in range(cleanup_size + 1):
+        vm_names.append('TEST_Provision_VM_{}'.format(i))
```

Comment:
A time stamp pre-fixing the name will help us determine in the event of multiple workloads that involve provisioning which workload "left behind" any vms or is currently provisioning vms so I think our vm name should be something like:
```python
 vm_name = '{}-provision'.format(time.strftime('%Y%m%d%H%M%S'))
```
I think we should let the value at the end grow to the number of vms we attempted to provision rather than wrap.  Thoughts?